main.tf

```
@@ -9,7 +9,7 @@ locals {
   }
 
   asp_name = "${var.asp_name != "null" ? var.asp_name : local.default_resource_group_name}"
-  asp_rg   = "${var.asp_rg != "null" ? var.asp_rg : local.default_resource_group_name}"
+  asp_rg   = "${var.asp_rg != "null" ? var.asp_rg : local.resource_group_name}"
```

this change should be fine to be mainlined if you separate it out to a different PR

---

If the intent is to declare a Resource Group externally to the module and pass it in as a module parameter then this change is can be removed in favour of:

```suggestion
  asp_rg   = "${var.asp_rg}"
```

---

unfortunately there's backwards compatibility to keep and we can't really make that change as it would be breaking